zio-http/src/test/scala/zio/http/model/headers/values/AcceptSpec.scala

```diff
@@ -0,0 +1,37 @@
+package zio.http.model.headers.values
+
+import zio.http.model.headers.values.Accept.{AcceptValue, InvalidAcceptValue}
+import zio.http.model.{MediaType, MimeDB}
+import zio.test._
+
+object AcceptSpec extends ZIOSpecDefault with MimeDB {
+  override def spec = suite("Accept header suite")(
+    test("parsing of invalid Accept values") {
+      assertTrue(Accept.toAccept("") == InvalidAcceptValue) &&
+      assertTrue(Accept.toAccept("something") == InvalidAcceptValue)
+    },
+    test("parsing of valid Accept values") {
+      assertTrue(Accept.toAccept("text/html") == AcceptValue(List(text.`html` -> None))) &&
+      assertTrue(Accept.toAccept("text/html;q=0.8") == AcceptValue(List(text.`html` -> Some(0.8)))) &&
+      assertTrue(Accept.toAccept("text/*") == AcceptValue(List(MediaType("text", "*") -> None))) &&
+      assertTrue(Accept.toAccept("*/*") == AcceptValue(List(MediaType("*", "*") -> None))) &&
+      assertTrue(Accept.toAccept("*/*;q=0.1") == AcceptValue(List(MediaType("*", "*") -> Some(0.1)))) &&
+      assertTrue(
+        Accept.toAccept("text/html, application/xhtml+xml, application/xml;q=0.9, */*;q=0.8") ==
+          AcceptValue(
+            List(
+              text.`html`             -> None,
+              application.`xhtml+xml` -> None,
+              application.`xml`       -> Some(0.9),
+              MediaType("*", "*")     -> Some(0.8),
+            ),
+          ),
+      )
+    },
+    test("parsing and encoding is symmetrical") {
+      (text.all ++ application.all).map { mediaType =>
```

Comment:
I think the issue here is that we are doing `reduce` which is handled by a recursive function, so the size of the dataset probably causes that to overflow. I think we can fix that on the `zio-test` side. In the meanwhile, can we map the whole list and then compare results? Something like :
```
val results = allMediaTypes.map(mediaType => Accept.fromAccept(Accept.toAccept(mediaType.fullType)))
assertTrue(allMediaTypes.map(_.fullType) == results) 
```